src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java

```diff
@@ -85,14 +102,41 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       commentEntity =
           CommentUtil.createRootEntity(lecture, timestampMs, author, content, dateNow, type);
     }
-
     datastore.put(commentEntity);
+
     response.setStatus(HttpServletResponse.SC_ACCEPTED);
   }
 
+  private Optional<String> validatePostRequest(HttpServletRequest request) {
+    if (request.getParameter(PARAM_LECTURE) == null) {
+      return Optional.of(ERROR_MISSING_LECTURE);
+    }
+    if (request.getParameter(PARAM_TYPE) == null) {
+      return Optional.of(ERROR_MISSING_COMMENT_TYPE);
+    }
+
+    Comment.Type type = Comment.Type.valueOf(request.getParameter(PARAM_TYPE));
+    if (type == Comment.Type.REPLY) {
```

Comment:
These nested if statements are not necessary. When you see nested if-statements you should try to re-write it to something simpler. 

if (type == Comment.Type.REPLY && request.getParameter(PARAM_PARENT) == null) {
  return error;
 }

if (request.getParameter(PARAM_TIMESTAMP) == null) {
  return error;
}

Reply:
Refactored